app-backend/database/src/main/scala/com/azavea/rf/database/tables/Chips.scala

```diff
@@ -0,0 +1,202 @@
+
+    for {
+      totalChips <- totalChipsQuery
+      chips <- paginatedChips
+    } yield {
+      val hasNext = (pageRequest.offset + 1) * pageRequest.limit < totalChips
+      val hasPrevious = pageRequest.offset > 0
+      PaginatedResponse[Chip](totalChips, hasPrevious, hasNext,
+        pageRequest.offset, pageRequest.limit, chips)
+    }
+  }
+
+  /** Delete a scene from the database
+    *
+    * @param chipId UUID ID of scene to delete
+    * @param user        Results will be limited to user's organization
+    */
+  def deleteChip(chipId: UUID, user: User)
+                     (implicit database: DB): Future[Int] = {
+
+    val action = Chips
+                   .filterToSharedOrganizationIfNotInRoot(user)
+                   .filter(_.id === chipId)
+                   .delete
+    logger.debug(s"Deleting chip with: ${action.statements.headOption}")
+    database.db.run {
+      action.map {
+        case 1 => 1
+        case 0 => 0
+        case c => throw new IllegalStateException(s"Error deleting chip: update result expected to be 1, was $c")
+      }
+    }
+  }
+
+  /** Update a chip in the database
+    *
+    * Allows updating the chip from a user -- does not allow a user to update
+    * createdBy or createdAt fields
+    *
+    * @param chip Chip scene to use to update the database
+    * @param chipId UUID ID of scene to update
+    */
+  def updateChip(chip: Chip, chipId: UUID, user: User)
+                     (implicit database: DB): Future[Int] = {
+
+    val updateTime = new Timestamp((new java.util.Date).getTime)
+
+    val updateChipQuery = for {
+      updateChip <- Chips
+                           .filterToSharedOrganizationIfNotInRoot(user)
+                           .filter(_.id === chipId)
+    } yield (
+      updateChip.modifiedAt, updateChip.x, updateChip.y,
+      updateChip.scene, updateChip.url, updateChip.labelProbabilities
+    )
+    database.db.run {
+      updateChipQuery.update((
+        updateTime, chip.x, chip.y,
+        chip.sceneId, chip.url, chip.labelProbabilities
+      )).map {
+        case 1 => 1
+        case c => throw new IllegalStateException(s"Error updating chip: update result expected to be 1, was $c")
+      }
+    }
+  }
+}
+
+class ChipDefaultQuery[M, U, C[_]](chips: Chips.TableQuery) {
+
+  def filterByChipParams(chipParams: ChipQueryParameters): Chips.TableQuery = {
+    chips
+      .filter(_.x === chipParams.x)
+      .filter(_.y === chipParams.y)
+      .filter(_.scene === chipParams.sceneId)
+      .filter { chip =>
+        chip.labelProbabilities.asInstanceOf[Json].as[Map[String, Float]] match {
+          case Left(failure) => true.asInstanceOf[Rep[Boolean]] // shouldn't happen
+          case Right(labelProbabilitiesMap) => {
+            decode[List[String]](chipParams.labels.get) match {
+              case Left(failure) => true.asInstanceOf[Rep[Boolean]] // shouldn't happen
+              case Right(labels) => {
```

Comment:
chipParams.labels can't be stored as a list of labels - rather, it is a string of the label list, and it needs to be decodeded